### PR TITLE
Pass tagged literal to csjs internally, allows short-syntax with single import

### PR DESCRIPTION
**Pull request by drcmda**
This allows react-csjs to be used as a decorator and literal tag at the same time. Since the decorator is the default export, it can be named "csjs," thereby removing the need for the additional "csjs" import, also postCSS transforms and syntax highlighting will work out of the box.

```
import csjs from 'react-csjs';

@csjs`
.figure {
    margin: 0;
    width: 100%;
    position: relative;
    min-height: 100vh;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    & > span {
        padding: 0 30%;
        font-size: 1.5em;
    }
}`
export default class Figure extends React.Component {
    render() {
```

This PR does not break the old way. Both will work. The HOC can still be exported as withStyles and receive a csjs obj. (-:

### CODE REVIEW

**Comment by tizmagik** [comment]
Thank you for this, this is great! I had no idea a decorator could also be a tagged template string, really neat!

With this, could you also update the README with this new interface? Especially to reflect your note from #6 : 

> About the peerDependency, i think since npm 3 they're not fetched automatically, but will return a warning if unresolved. Is this wanted? As long as the readme says
> `npm install --save csjs react-csjs`
> that would be fine. (Right now it just mentions react-csjs)

I'll try and spend some time later this week playing around with this PR to see how it affects the interface, if at all, and get back to you. Thanks again!

**Comment by drcmda** [comment]
Readme updated. Let me know if it needs changes.

**Comment by tizmagik** [comment]
Hey @drcmda thanks again for this, I really want to merge this but I was hoping you can help me think through a few details first. I'm concerned that it could be somewhat confusing to users when to use

``` js
import csjs from 'csjs';
```

vs

``` js
import csjs from 'react-csjs';
```
- Why do I need `csjs` at all since It looks like `react-csjs` provides it?
- Or, maybe we should keep the decorator named as `withStyles` in the docs so that it can be used like this?

``` jsx
import withStyles from 'react-csjs';

@withStyles`
  .foo {
    color: red;
  }
`
export default class Thing extends Component {
  render() {
    return <div className={this.props.classes.foo} />
  }
}
```

**Comment by tizmagik** [comment]
Also, please be sure to add yourself as a contributor in package.json -- you'll be getting commit access once this PR is merged (my rule is 2 contributions grants commit privileges) 😄

File: README.md
Comment (tizmagik) [inline README.md]: I think we should promote this style first, so the "Or" case is importing `csjs` explicitly

File: README.md
Comment (tizmagik) [inline README.md:15]: Add a note above this:

```
You'll need to install both `csjs` and `react-csjs` like so:
```

File: README.md
Comment (tizmagik) [inline README.md]: Does this actually require a semicolon here?

File: src/index.js
Comment (tizmagik) [inline src/index.js:12]: Instead of modifying the `css` param, can we rename it in the signature and then use `css` locally here?

**Review by tizmagik — CHANGES_REQUESTED** [review]
A few minor tweaks + some thoughts on how we can make the README a bit clearer about when to use `csjs` directly vs just `react-csjs`

**Comment by tizmagik** [comment]
Also, how were you testing this? I ask because #12 is landing soon and #14 might be useful to you.

**Comment by drcmda** [comment]
@tizmagik 

For clarity, this is how the example on the main page would look like

``` javascript
import React from 'react';
import csjs  from 'react-csjs';

@csjs`
.button {
  background-color: purple
}
.label {
  color: blue
}`
export default class Button extends React.Component {
  render() {
    return (
      <div className={this.props.classes.button}>
        <span className={this.props.classes.label}>
          {this.props.children}
        </span>
      </div>
    )
  }
}
```

> Why do I need csjs at all since It looks like react-csjs provides it?

We don't need to import csjs at all, that's the cool thing about it :) 

> maybe we should keep the decorator named as withStyles in the docs so that it can be used like this?

The problem was that this will break postCSS and syntax highlighting. 3rd party plugs are hardcoded to the "csjs" signature and wouldn't work with "withStyles" 

The good news is, react-csjs is a default export and they are nameless. The actual name gets defined by the import statement. But since we can now pass literals, we can skip the _original csjs import_ and  decotare classes with csjs. We thereby profit from the babel passes (editor plugs and so on) and get the smallest possible footprint. :)

But nothing changes with this. It is still optional.

File: src/index.js
Comment (drcmda) [reply src/index.js:12]: css is later passed to 

```
this.elm = insertStyle(csjs.getCss(css));
```

which reads the csjs object the function normally receives.  This would extend it to string literals, which are converted to the csjs object which is expected later.

File: README.md
Comment (drcmda) [reply README.md]: I think it's safe to remove it 👍

**Comment by drcmda** [comment]
I made the additions. I also opted for a little shorter CSS to keep examples tidy.

**Comment by tizmagik** [comment]
> We don't need to import csjs at all, that's the cool thing about it :)

Right, I get that. What I'm saying is it wasn't completely clear to me as a user reading the README. But I think re-ordering the usage examples and some of the explanatory texts you added helped a lot. Thanks!

> The problem was that this will break postCSS and syntax highlighting. 3rd party plugs are hardcoded to the "csjs" signature and wouldn't work with "withStyles"

Ah, great point! 👍

File: src/index.js
Comment (tizmagik) [reply src/index.js:12]: Right, but this breaks a lint rule `no-param-reassign`. If you lint this branch it should complain about this line (if it doesn't, something is wrong with the linter), so I suggest to do something like:

``` js
export default function (userCss, values) {
  let css = userCss;

  if (Array.isArray(css)) {
  ...
```

**Comment by tizmagik** [comment]
Labeling this as a `breaking change` since we'll no longer export a named `{withStyles}` function -- which I think is for the better.

**Comment by drcmda** [comment]
@tizmagik the function name only plays a role internally, within the module. If you need to add prototypes to it etc. Since it is the default export outside it is always overridden by the import clause.

**Comment by drcmda** [comment]
@tizmagik Looks good.

File: src/index.js
Comment (drcmda) [reply src/index.js:12]: oh, wasn't aware of this ... will update asap

**Comment by tizmagik** [comment]
One last thing then I think we can merge this. Can you add a test for using the tagged decorator syntax please?

**Comment by drcmda** [comment]
@tizmagik tbh, i wouldn't know where to start. I haven't done much testing. Could we merge and ask @wKovacs64 ?

**Comment by tizmagik** [comment]
Sure, let's do that. I'll create an issue for a follow-on PR.

Thanks for this, great work!!

**Review by tizmagik — APPROVED** [review]

**Pull request merged**
